malaria-ios/ViewControllers/PlanTrip/PlanTripViewController.swift

```diff
@@ -316,4 +354,63 @@ extension PlanTripViewController {
     private var AlertOptions: (ok: String, cancel: String) {get {
         return ("Ok", "Cancel")
         }}
-}
\ No newline at end of file
+}
+
+
+//Google Autocomplete Delegate function implementation
+extension PlanTripViewController: GMSAutocompleteViewControllerDelegate {
+    
+    // Handle the user's selection.
+    func viewController(viewController: GMSAutocompleteViewController, didAutocompleteWithPlace place: GMSPlace) {
+        generateTripBtn.enabled = true
+        tripLocation = "value"
+        if let loc:String = "\(place.name)" {
+            location.text = loc
+            tripLocation = loc
+            generateTripBtn.enabled = true
+        }
+        self.dismissViewControllerAnimated(true, completion: nil)
+    }
+    
+    func viewController(viewController: GMSAutocompleteViewController, didFailAutocompleteWithError error: NSError) {
+        Logger.Error("Error: ", function: error.description)
+    }
+    
+    func wasCancelled(viewController: GMSAutocompleteViewController) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+        Logger.Info("User canceled the operation")
+    }
+    
+    func didRequestAutocompletePredictions(viewController: GMSAutocompleteViewController) {
+        UIApplication.sharedApplication().networkActivityIndicatorVisible = true
+        Logger.Info("Turn the network activity indicator on and off again.")
+    }
+    
+    func didUpdateAutocompletePredictions(viewController: GMSAutocompleteViewController) {
+        UIApplication.sharedApplication().networkActivityIndicatorVisible = false
+    }
+    
+}
+
+// Handle the user's selection.
+extension PlanTripViewController: GMSAutocompleteResultsViewControllerDelegate {
+    func resultsController(resultsController: GMSAutocompleteResultsViewController,
+        didAutocompleteWithPlace place: GMSPlace) {
+            searchController?.active = false
+    }
+    
+    func resultsController(resultsController: GMSAutocompleteResultsViewController,
+        didFailAutocompleteWithError error: NSError){
+            Logger.Error("Error: ", function: error.description)
```

Comment:
Just this ;) 
Logger.Error(error.description)

If you notice in the Logger filem the function is the name of the function that called the method:
```func Info(message: CustomStringConvertible, function: String = __FUNCTION__, path: String = __FILE__, line: Int = __LINE__)```

You just need to provide something that is convertible to a string ;)